File: data_updates/R/iati_rhfp2.R
####Setup###/#
list.of.packages <- c("reshape2","data.table","openxlsx","plyr","gdata","varhandle","rsdmx","rstudioapi","RCurl")
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])]
if(length(new.packages)) install.packages(new.packages)
lapply(list.of.packages, require, character.only=T)

setwd(dirname(getActiveDocumentContext()$path))

# bring in data
options(timeout = 1000)
temp <- tempfile()
download.file("https://ddw.devinit.org/api/export/1271",temp,method = "libcurl")
dat <- fread(temp)
names(dat) = c("Code type","Year","recipient_name","Reporting Organisation Narrative", "disbursed")

#replace sector code with sector name
dat$'Code type'[which(dat$`Code type`==13020)] = "Reproductive health care"
dat$'Code type'[which(dat$`Code type`==13030)] = "Family planning"

#create 13020 + 13030
AggData = aggregate(dat$disbursed, by=list(dat$`Reporting Organisation Narrative`,dat$recipient_name,dat$Year), FUN=sum)
names(AggData) = c("Reporting Organisation Narrative","recipient_name","Year","disbursed")
AggData$'Code type'<- "Reproductive health care and family planning"

#combine dataframes
dat3 = rbind(AggData,dat)

#Change to wide format
datwide = dcast(dat3, `Code type` + `Reporting Organisation Narrative` + `recipient_name` ~ `Year`,value.var="disbursed") 

write.csv(datwide,"iati_rhfp2.csv")

File: data_updates/Python/iati_rhfp2.py
import os
import requests
from requests.auth import HTTPBasicAuth
import pandas as pd


print("Starting read-in")

data = pd.read_csv("https://ddw.devinit.org/api/export/1271")

data = pd.DataFrame(data)

data.columns = ["Code type","Year", "recipient_name","Reporting Organisation Narrative", "disbursed"]

data = data[(data['Code type'] != 13010) & (data['Code type'] != 13081)]

# Sector code mapping

data.loc[data['Code type']==13020,'Code type'] = "Reproductive health care"
data.loc[data['Code type']==13030,'Code type'] = "Family planning"

# Summing both

total_data = data.groupby(["Year","recipient_name","Reporting Organisation Narrative"]).agg({"disbursed":"sum"}).reset_index()
total_data['Code type'] = "Reproductive health care and family planning"

data = data.append(pd.DataFrame(data = total_data),ignore_index=True)

# Put it in millions

data["disbursed"] = data["disbursed"]/1000000

data.to_csv("iati_rhfp2.csv", encoding='utf-8', index=False)